[Justin Yip] iP

justinyjt opened the pull request:
### PaggroBot
> "I never took a day off in my 20s. Not one." - Bill Gates ([source](https://www.keepinspiring.me/hard-work-quotes/))

PaggroBot pushes you to work hard and complete the tasks that you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](url).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😒 

And it is **FREE!**

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon) 

If you are a Java programmer, you can use it to practice Java too! Here's the main method:
```java
public static void main(String[] args) throws IOException {
        new PaggroBot("../../../data/paggro.txt").run();
}
```

Commit:
Add Gradle support

justinyjt committed:
Create skeletal version of PaggroBot

justinyjt committed:
Change filename from Duke to PaggroBot

justinyjt committed:
Remove deprecated Duke file

justinyjt committed:
Implement new class Lister with add and list methods

justinyjt committed:
Implement mark and unmark to represent task completion

justinyjt committed:
Implement Task class instead of using primitive values

justinyjt committed:
Fix bug where program only correct for single word input. Add 3 new subclasses of Task.

justinyjt committed:
Set up test script with sample input and expected output

justinyjt committed:
Implement error checking and handling

justinyjt committed:
Create PaggroException class for error handling

justinyjt committed:
Implement delete method by changing Task array to ArrayList

justinyjt committed:
Add data folder and allow saving of task list data

justinyjt committed:
Add NotableDate class and new cmd listOnDate

justinyjt committed:
Add Time to deadline and event

justinyjt committed:
Merge branch Level 7 and 8

justinyjt committed:
Add Command, Storage, Parser, Ui classes for encapsulation of tasks

justinyjt committed:
Add packages, extract Command classes, add JUnit testing

justinyjt committed:
Package all classes, add Command classes and add JUnit tests

justinyjt committed:
Add data folder

justinyjt committed:
Add JavaDoc comments

justinyjt committed:
Remove coding standard violations

justinyjt committed:
Add find command

justinyjt committed:
Merge JavaDoc and CodingStandard branches

justinyjt committed:
Merge Level-9 branch to master

justinyjt committed:
Add gradle support

justinyjt committed:
Add graddle build

justinyjt committed:
Change file path relative to local

justinyjt committed:
Merge with branch-A-Gradle

justinyjt committed:
Add checkstyle and fix style errors

justinyjt committed:
Add GUI using JavaFX and FXML

justinyjt committed:
Change JavaFX in FXML doc from 17 to 11

nicollegann commented on src/main/java/paggro/command/ByeCommand.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    /**
+     * Carries out the execution of the Bye command which terminates PaggroBot.
+     * @param lister The Lister object for the command to execute on.
```
Should there be a one-line buffer between parameters and the description? I think this is repeated in several other java-docs too.

justinyjt replied:
Yup, you are right. Thanks for pointing that out :)

nicollegann commented on src/main/java/paggro/command/Command.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * The arguments of a command, if applicable
+     */
+    String parameters;
```
Might be better to declare `String parameters` as `private`

nicollegann commented on src/main/java/paggro/storage/Storage.java:
```diff
@@ -0,0 +1,221 @@
+                }
+            }
+            return new Lister(tasks, dateMap);
+        }
```
I like how the switch-case statements follows the correct form! It also made the code very neat and easy to read 😄

nicollegann commented on src/main/java/paggro/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,78 @@
+            throw new PaggroException("    Could not add to paggro.txt =.=");
+        }
+
+        return ui.showAdded(task) + "\n" + ui.showNumber(lister.getTasks().size());
```
I really like how your code is separated into logical blocks and this is consistent throughout all the similar classes! 👍

nicollegann reviewed (commented):
Overall, I found your code very neat and easy to read! There aren't any major violations of Java coding standards and the Javadocs are very well written 💯 Perhaps the only thing to take note of is leaving an empty line between the description and parameter section 😅

tsumian commented on src/main/java/paggro/command/DeleteCommand.java:
```diff
@@ -0,0 +1,75 @@
+        try {
+            index = Integer.parseInt(this.getParameters());
+        } catch (NumberFormatException e) { // parameter was not a number
+            throw new PaggroException("    Really? Can you input an actual number this time... =.=");
```
Perhaps you could abstract the spacings/ tabs out as a variable so that you do not need to manually put in the spacings whenever you output an error message.

tsumian commented on src/main/java/paggro/notabledate/NotableDate.java:
```diff
@@ -0,0 +1,59 @@
+import paggro.task.Task;
+
+/**
+ * This class encapsulates a NotableDate object which contains a list of all the tasks on a given date.
```
I like how succinct and efficient your JavaDoc comments are. Can be seen throughout most classes. Good job!

tsumian commented on src/main/java/paggro/task/Deadline.java:
```diff
@@ -0,0 +1,84 @@
+        } else {
+            return "D | " + Boolean.toString(isDone()) + " | " + getDescription() + " | "
+                    + date.getLocalDate().format(dFormat) + " | " + time.format(tFormat);
+        }
```
Good formatting to enhance readability.

tsumian commented on src/main/java/paggro/lister/Lister.java:
```diff
@@ -0,0 +1,118 @@
+     * @param key String of the keyword to search for.
+     * @return ArrayList of tasks with keyword.
+     */
+    public ArrayList<Task> findIndicesContaining(String key) {
```
I like how most of your variable names and method names are explanatory. The naming is sufficient for reader to know what is going on in your general code.

tsumian reviewed (commented):
Overall really good naming across most classes. Although there were a little bit of inconsistencies in style and perhaps some redundancies that could be abstracted away, it is a good and readable piece of code! Good Job!

justinyjt committed:
Modify JavaDocs for methods as suggest in PR review

justinyjt committed:
Add welcome message to GUI

justinyjt committed:
Add Java assertions

After adding a task to the task list of the Lister object,
it is expected that the size of the task list is more than zero.

The size of the task list should be checked after commands like
ToDo, Event and Deadline to ensure that the program is working as
intended.

Let's add assert statements to each of these commands to confirm
that the task list contains at least 1 item after their execution.

justinyjt committed:
Merge pull request #1 from justinyjt/branch-A-Assertions

Add Java assertions

justinyjt committed:
Remove code that violates code quality

Remove magic literals from code by creating local
or static variables for them to improve code quality.

justinyjt committed:
Merge pull request #2 from justinyjt/branch-A-CodeQuality

Remove code that violates code quality

justinyjt committed:
Set up CI using GitHub Actions

justinyjt committed:
Does nothing, dummy commit to test CI

justinyjt committed:
Fix checkstyle errors and build tests

justinyjt committed:
Dummy commit for CI

justinyjt committed:
Merge branch-A-CI

justinyjt committed:
Add tagging functionality

justinyjt committed:
Reduce long methods using SLAP

justinyjt committed:
Improve GUI

justinyjt committed:
Fix checkstyle for build

justinyjt committed:
Update README.md

justinyjt committed:
Add README.md

justinyjt committed:
Update README.md

justinyjt committed:
Update README.md
